File: 7kyu/greatest-common-divisor/greatest-common-divisor.js
/* 
note:
use Euclidean algorithm
https://www.freecodecamp.org/news/how-to-use-the-euclidean-algorithm-to-find-the-greatest-common-divisor-gcd/
*/

function mygcd(x, y) {
	let r;
	while (x % y > 0) {
		r = x % y;
		x = y;
		y = r;
	}
	return y;
}

File: 7kyu/greatest-common-divisor/greatest-common-divisor.go
package main

func Gcd(x, y uint32) uint32 {
	if y == 0 {
		return x
	}

	return Gcd(y, x%y)
}

// func Gcd(x, y uint32) uint32 {
// 	for y != 0 {
// 		x, y = y, x%y
// 	}

// 	return y
// }
